bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandlerFactory.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.senechome.internal;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link SenecHomeHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Steven Schwarznau - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.senechome", service = ThingHandlerFactory.class)
+public class SenecHomeHandlerFactory extends BaseThingHandlerFactory {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections
+            .singleton(SenecHomeBindingConstants.THING_TYPE_SENEC_HOME_BATTERY);
+
+    private final SenecHomeApi senecHomeApi;
+
+    @Activate
+    public SenecHomeHandlerFactory(@Reference HttpClientFactory httpClientFactory) {
+        this.senecHomeApi = new SenecHomeApi(httpClientFactory.getCommonHttpClient());
```

Comment:
Hmm It realize the hostname is specific for each thing. So this means the hostname is different on each hostname and thus the `SenecHomeApi` should not be created here, but it the handler itself. So here it should just store the `httpClient` in a field, pass that field to the handler:

```suggestion
        this.httpClient = httpClientFactory.getCommonHttpClient();
```

and in the handler create the `SenecHomeApi` like in the handler constructor, but with the `httpClient` passed in.

```
public SenecHomeHandler(Thing thing, HttpClient httpClient) {
        super(thing);
        this.senecHomeApi = new SenecHomeApi(httpClient);
```